nvflare/apis/executor.py

```diff
@@ -49,3 +50,26 @@ def execute(self, task_name: str, shareable: Shareable, fl_ctx: FLContext, abort
             An output shareable.
         """
         pass
+
+    def get_job_target_type(self):
+        """This method is required by Job API.
+
+        Returns:
+
+        """
+        return JobTargetType.CLIENT
+
+    def add_to_fed_job(self, job, ctx, tasks=None):
+        """This method is required by Job API.
+
+        Args:
+            job: the Job object to add to
+            ctx: Job Context
+            tasks: tasks for the executor
+
+        Returns:
+
+        """
+        if not tasks:
+            tasks = ["*"]
+        job.add_executor(obj=self, tasks=tasks, ctx=ctx)
```

Comment:
Why do we choose to add this method to all other existing abstracts of base components?

I prefer the wrapper idea better (like what you do with TFModel and PTModel) as it does not affect these existing base classes.

Reply:
We originally had the JobObj wrapper idea, however we decided to go with this duck typing method to remove the need for a wrapper. You are right, it is a tradeoff between user experience and affecting existing base classes.